lightning/src/util/mod.rs

```diff
@@ -47,3 +47,21 @@ pub mod test_utils;
 /// machine errors and used in fuzz targets and tests.
 #[cfg(any(test, feature = "fuzztarget", feature = "_test_utils"))]
 pub mod enforcing_trait_impls;
+
+use io;
+use prelude::*;
+
+/// core2 doesn't have read_to_end, so implement it ourselves
+pub fn read_to_end<D: io::Read>(mut d: D) -> Result<Vec<u8>, io::Error> {
```

Comment:
Why does this need to be pub? Also, shouldn't it go in `io_extras`? It feels really random shoved into a top-level `mod.rs`?

Reply:
I was a bit uncomfortable putting it in `io_extras` because it was `no_std` only.  But I added an `std` `io_extras` that forwards it to the `std` impl, and it seems fine